activerecord/test/cases/adapters/mysql2/schema_migrations_test.rb

```diff
@@ -11,6 +11,14 @@ def test_renaming_index_on_foreign_key
     connection.remove_foreign_key :engines, name: "fk_engines_cars"
   end
 
+  def test_removing_column_with_foreign_key_deletes_foreign_key
```

Comment:
Let's put the test somewhere it'll run on all adapters: MySQL is getting a helping hand, but we ultimately intend the behaviour from all of them.